assemblies/features/base/src/main/filtered-resources/resources/bin/karaf

```diff
@@ -93,7 +93,12 @@ checkRootInstance() {
           if ps -p "${ROOT_INSTANCE_PID}" > /dev/null
           then
               MAIN=org.apache.karaf.main.Main
-              PID_COMMAND=$(ps -p "${ROOT_INSTANCE_PID}" -o args | sed 1d)
+              DIST_OS=`uname -s | tr [:upper:] [:lower:] | tr -d [:blank:]`
+
+              case "$DIST_OS" in
+                'hp-ux')PID_COMMAND=$(UNIX95= ps -p "${ROOT_INSTANCE_PID}" -o args | sed 1d);;
+                *)PID_COMMAND=$(ps -p "${ROOT_INSTANCE_PID}" -o args | sed 1d);;
+              esac
               if [ "${PID_COMMAND#*$MAIN}" != "$PID_COMMAND" ]; then
```

Comment:
Would you mind leveraging and extending the `detectos` function instead ? https://github.com/apache/karaf/blob/master/assemblies/features/base/src/main/filtered-resources/resources/bin/inc#L28-L57

Reply:
As requested, we updated the `detectos` function to add a specific variable to be associated with the ps command on HP-UX platforms when the `-o` option is used.

This was not mentioned on the [KARAF-5753](https://issues.apache.org/jira/browse/KARAF-5753) but we also provide an improvement for the following line in shell scripts:

 `if [ -z $READLINK_EXISTS ]; then`

**$READLINK_EXISTS** was not protected, causing error message on some platforms where the variable was empty (see below):

`client[2]: test: Specify a parameter with this command.`
`client[17]: readlink:  not found.`